source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -28,6 +28,59 @@ namespace Extensions {
 namespace StatSinks {
 namespace Common {
 namespace Statsd {
+// clang-format off
+// SPELLCHECKER(off)
+//
+// Performance benchmarks show this is slightly faster as an array of uint32_t
+// rather than an array of bool.
+static constexpr uint32_t needs_statsd_sanitizer[256] = {
+  // Control-characters 0-31.
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  // Pass through printable characters starting with space.
+  0, 0, 0, 1, 0, 0, 0, 0,                 //  !"#$%&'
+  0, 0, 0, 0, 0, 0, 0, 0,                 // ()*+,-.7
+  0, 0, 0, 0, 0, 0, 0, 0,                 // 01234567
+  0, 0, 1, 0, 0, 0, 0, 0,                 // 89:;<=>?
+  1, 0, 0, 0, 0, 0, 0, 0,                 // @ABCDEFG
+  0, 0, 0, 0, 0, 0, 0, 0,                 // HIJKLMNO
+  0, 0, 0, 0, 0, 0, 0, 0,                 // PQRSTUVW
+  0, 0, 0, 0, 0, 0, 0, 0,                 // XYZ[\]^_
+  0, 0, 0, 0, 0, 0, 0, 0,                 // `abcdefg
+  0, 0, 0, 0, 0, 0, 0, 0,                 // hijklmno
+  0, 0, 0, 0, 0, 0, 0, 0,                 // pqrstuvw
+  0, 0, 0, 0, 1, 0, 0, 0,                 // xyz{|}~\177
+
+  // 0x80-0xff, all of which require pass through.
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+};
+// SPELLCHECKER(on)
+// clang-format on
+
+absl::string_view sanitizeStatsdName(absl::string_view name, std::string& buffer) {
+  static_assert(ARRAY_SIZE(needs_statsd_sanitizer) == 256);
+  uint32_t need_statsd_sanitization = 0;
+  for (char c : name) {
+    need_statsd_sanitization |= needs_statsd_sanitizer[static_cast<uint8_t>(c)];
+  }
+  if (need_statsd_sanitization == 0) {
+    return name; //  Fast path, should be executed most of the time.
+  }
+  buffer = name;
+  for (char& c : buffer) {
+    if (needs_statsd_sanitizer[static_cast<uint8_t>(c)]) {
+      c = '_';
```

Comment:
There's a functional difference here with the sanitizer that will replace "://" with a single "_" is that right? Could that be related to the test changes you spoke of earlier?

Reply:
yes there is a functional change in behavior. Some tests failure on enabling this feature are attributed to this. I am working on fixing the impacted tests.

Reply:
I don't think we need there to be a change in behavior for this. I think we can have the fast-path to check whether anything needs to be done, and if it does, just use absl:StrReplaceAll rather than changing behavior.

I just want to have the "nothing needs to be done" check be fast.